scripts/abi_to_rc/abi_to_rc.py

```diff
@@ -0,0 +1,157 @@
+            sys.exit(0)
+        elif "rc-action-template.md" != sys.argv[3]:
+            print("ERROR: third argument should be the 'rc-action-template.md' file")
+            print("-- Expected format --")
+            print("$ python abi_to_rc.py ${{smart-contract.abi}} rc-overview-template.md rc-action-template.md")
+            sys.exit(0)
+        elif not os.path.isfile(sys.argv[3]):
+            print("ERROR: %s could not be found" % sys.argv[3])
+            sys.exit(0)
+    else:
+        print("ERROR: too many arguments")
+        print("-- Expected format --")
+        print("$ python abi_to_rc.py ${{smart-contract.abi}} rc-overview-template.md rc-action-template.md")
+        sys.exit(0)
+
+# gets actions, inputs and input types from abi file
+def get_actions_inputs_types():
+    abi_file_name = sys.argv[1]
+    abi_file = open(abi_file_name,'r')
+    abi_text = abi_file.read()
+    abi_file.close()
+    abi_json = json.loads(abi_text)
+    actions_json = abi_json['actions']
+    for obj in actions_json:
+        actions.append(obj)
+    structs_json = abi_json['structs']
+    for action in actions:
+        inputs[action['name']] = []
+        types[action['name']] = []
+    for struct in structs_json:
+        for action in actions:
+            if struct['name'] == action['type']:
+                for field in struct['fields']:
+                    inputs[action['name']].append(field['name'])
+                    types[action['name']].append(field['type'])
+
+# builds rows for the table
+def build_table_rows():
+    table_rows = []
+    row = "| `${{ action }}` | `${{ input }}` | `${{ type }}` |"
+    for action in actions:
+        temp_row = row
+        temp_row = temp_row.replace("${{ action }}",action['name'])
+        if len(inputs[action['name']]) > 1:
+            input_string = ""
+            for name in inputs[action['name']]:
+                input_string = input_string + "`" + name + "`" + "<br/>"
+            temp_row = temp_row.replace("`${{ input }}`",input_string)
+        elif len(inputs[action['name']]) == 1:
+            temp_row = temp_row.replace("${{ input }}",inputs[action['name']][0])
+        else:
+            temp_row = temp_row.replace("${{ input }}",action['type'])
+
+        if len(types[action['name']]) > 1:
+            type_string = ""
+            for name in types[action['name']]:
+                type_string = type_string + "`" + name + "`" + "<br/>"
+            temp_row = temp_row.replace("`${{ type }}`",type_string)
+        elif len(types[action['name']]) == 1:
+            temp_row = temp_row.replace("${{ type }}",types[action['name']][0])
+        else:
+            temp_row = temp_row.replace("${{ type }}",action['type'])
+
+        table_rows.append(temp_row)
+    return table_rows
+
+# generates an overview ricardian contract from the overview template
+def generate_rc_overview_file():
+    tr = build_table_rows()
+    rc_overview_template_file_name = sys.argv[2]
```

Comment:
To avoid the documented problem of calling the script while outside the script's own directory and being unable to find the template, derive a path: `os.path.join(os.path.dirname(sys.argv[0]), _RC_OVERVIEW)`.  (You did make a 'constant' of the template filename, right?)